core/src/test/java/org/apache/accumulo/core/data/KeyBuilderTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.data;
+
+import org.apache.hadoop.io.Text;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class KeyBuilderTest {
```

Comment:
Nice test.  Very thorough.  I tried running code coverage and everything in key builder was covered.

I think it would be useful to test the following.  This ensures that the behavior of calling timestamp multiple times is well defined going forward and does not change to take the first.

```java
k = Key.builder().row("r").timestamp(44).timestamp(99).build();
assertEquals(new Key("r","","",99l), k);
```

It would be nice to test that all builder methods fail fast when passed null.   I think they all will at the moment.   Want to avoid the case where null passed to earlier method causes failure in `build()` method and user does not know if the row, family, or qualifier was null.